### ISSUE TITLE
Search throws error Nil

### ISSUE THREAD

**Issue by dfpena** [dblock/slack-google-bot#3]
I am using slack-google-bot deployed on heroku
version : 0.1.0
Everything seems to be working except google search
entering bot name after joining channel displays giphy and version number
entering bot with : results in giphy bad command notification

entering botname [search expression]
gives error
undefined method `title' for nil:NilClass

I have tried using custom google search id and double checked that the custom search returned items for the search parameters.
Ideas?

**Comment by dblock** [comment]
Haven't touched this for a while. The library used here is using an API that Google deprecated, so the library needs to be swapped for something better/different. At the very least you could remove it and do something like [this](https://github.com/dblock/semi-intelligent-slack-bot-in-ruby/blob/master/06-intelligent-bot.rb).

``` ruby
results = JSON.parse HTTP.get("https://www.googleapis.com/customsearch/v1", params: {
      q: expression,
      key: ENV['GOOGLE_API_KEY'],
      cx: ENV['GOOGLE_CSE_ID']
    })
    result = results['items'].first
    next unless result
    message = result['title'] + "\n" + result['link']
```

I'd love a PR that fixes this project up.

**Comment by dfpena** [comment]
LOL Same here! I don't develop in ruby or I would have taken a crack it.

**Comment by dblock** [comment]
Looks like there's a fork with this fix, https://github.com/minhdanh/slack-google-bot/commit/9ae5e44f2630a6f46e4f8cb32f78f73184e2df00, maybe @minhdanh can PR those changes?

**Comment by minhdanh** [comment]
Sure. I'll PR the fix.

**Comment by jandreaucodes** [comment]
Even using @minhdanh's fork, I'm still getting `undefined method 'first' for nil:NilClass` with and without a CSE ID.

**Comment by dblock** [comment]
Find the call stack @mikeatdesk.

**Comment by jandreaucodes** [comment]
@dblock Can you elaborate? I'm not sure what your reply means.

**Comment by dblock** [comment]
I've fixed this on master by finishing #4. 

Note that Google now requires that you have an API key, that's something you need to create from https://console.developers.google.com/apis and set as GOOGLE_API_KEY. There also have been multiple iterations of bot infrastructure versions since this project was last updated, so this might not be your last problem. 

If the README/DEPLOYMENT docs aren't clear enough to get this bot running, please contribute!

The error comes from somewhere. When you have an error like this find the call stack, ie. debug it until you know why it throws it and from where.
